src/dags/parkeervakken.py

```diff
@@ -0,0 +1,357 @@
+            except Exception as e:
+                raise Exception(
+                    "Failed to create parkeervakken: {}".format(str(e)[0:150])
+                )
+        if len(regimes_sql):
+            try:
+                connection.execute(create_regimes_sql)
+            except Exception as e:
+                raise Exception("Failed to create regimes: {}".format(str(e)[0:150]))
+
+    return "Created: {} parkeervakken and {} regimes".format(
+        len(parkeervakken_sql), len(regimes_sql)
+    )
+
+
+def find_export_date():
+    """
+    Find closest export date.
+    """
+    today = datetime.date.today()
+    date = today
+    if today.weekday() == 3:
+        date = today
+    elif today.weekday() > 3:
+        # This week
+        date = today - datetime.timedelta(days=today.weekday() - 3)
+    else:
+        # Last week
+        date = today - datetime.timedelta(days=today.weekday() + 4)
+    return date.strftime("%Y%m%d")
+
+
+def run_imports(last_date):
+    """
+    Run imports for all files in zip that match date.
+    """
+    ids = []
+    for shp_file in source.glob("*{}.shp".format(last_date)):
+        duplicates = import_data(str(shp_file), ids)
+
+        if len(duplicates):
+            print("Duplicates found: {}".format(", ".join(duplicates)))
+
+
+with DAG(dag_id, default_args=default_args, description="Parkeervakken") as dag:
+    last_date = find_export_date()
+    zip_file = "nivo_{}.zip".format(last_date)
+    source = pathlib.Path(TMP_DIR)
+
+    mk_tmp_dir = BashOperator(task_id="mk_tmp_dir", bash_command=f"mkdir -p {TMP_DIR}")
+
+    fetch_zip = SwiftOperator(
+        task_id="fetch_zip",
+        container="tijdregimes",
+        object_id=zip_file,
+        output_path=f"{TMP_DIR}/{zip_file}",
+        conn_id="parkeervakken",
+    )
+
+    extract_zip = BashOperator(
+        task_id="extract_zip",
+        bash_command=f'unzip -o "{TMP_DIR}/{zip_file}" -d {TMP_DIR}',
+    )
+
+    create_temp_tables = PostgresOperator(
+        task_id="create_temp_tables",
+        sql=SQL_CREATE_TEMP_TABLES,
+        params=dict(base_table=f"{dag_id}_{dag_id}"),
+    )
+
+    run_import_task = PythonOperator(
+        task_id="run_import_task",
+        op_kwargs={"last_date": last_date},
+        python_callable=run_imports,
+        dag=dag,
+    )
+
+    rename_temp_tables = PostgresOperator(
+        task_id="rename_temp_tables",
+        sql=SQL_RENAME_TEMP_TABLES,
+        params=dict(base_table=f"{dag_id}_{dag_id}"),
+    )
+
+(
+    mk_tmp_dir
+    >> fetch_zip
+    >> extract_zip
+    >> create_temp_tables
+    >> run_import_task
+    >> rename_temp_tables
+)
+
+
+# Internals
+def create_parkeervaak(row):
+    geom = "''"
+    if row.shape.shapeTypeName == "POLYGON":
+        geom = "ST_GeometryFromText('{}', 28992)".format(str(Polygon(row.shape.points)))
+    sql = (
+        "("
+        "'{parkeer_id}',"
+        "'{buurtcode}',"
+        "E'{straatnaam}',"
```

Comment:
Waar is de 'E' voor de straatnaam voor bedoeld?

Reply:
Escapes sequences voor de straat namen met `'`.

Reply:
Ah, postgres dingetje dat ik nog niet kende (heb veel MySQL gedaan laatste jaren). Handig.